code/z_adventurezones/space_casino.dm

```diff
@@ -0,0 +1,371 @@
+
+	var/possible_vessels = list("/obj/item/reagent_containers/food/drinks/drinkingglass", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/shot", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/oldf", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/round", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/wine", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/cocktail", \
+							"/obj/item/reagent_containers/food/drinks/drinkingglass/flute")
+
+	var/possible_stuffs = list("/obj/item/cocktail_stuff/drink_umbrella", \
+							"/obj/item/cocktail_stuff/maraschino_cherry", \
+							"/obj/item/cocktail_stuff/cocktail_olive", \
+							"/obj/item/cocktail_stuff/celery")
+
+	var/possible_wedges = list("/obj/item/reagent_containers/food/snacks/plant/orange/wedge", \
+	                        "/obj/item/reagent_containers/food/snacks/plant/lime/wedge", \
+							"/obj/item/reagent_containers/food/snacks/plant/lemon/wedge", \
+							"/obj/item/reagent_containers/food/snacks/plant/grapefruit/wedge")
+
+	New()
+		..()
+		// Start by getting a few initial things
+		if (!src.home || !isturf(src.home))
+			src.home = get_turf(src)
+		if (src.homeTables.len == 0)
+			for (var/obj/table/reinforced/bar/T in view(5, src.home))
+				src.homeTables |= T
+
+	proc/get_empty_tables()
+		if (src.homeTables.len == 0)
+			for (var/obj/table/reinforced/bar/T in view(5, src.home))
+				src.homeTables |= T
+			if (src.homeTables.len == 0)
+				explode()
+		for (var/obj/table/reinforced/bar/T in src.homeTables)
+			if (!istype(T))
+				src.homeTables -= T
+			var/glasses = 0
+			for (var/obj/item/reagent_containers/food/drinks/drinkingglass in view(0, get_turf(T)))
+				glasses++
+			if (glasses < 3)
+				src.targets |= T
+
+	process()
+		// Nothing to do. Let's find something to do.
+		if (targets.len == 0)
+			get_empty_tables()
+			if (targets.len == 0) // No work to be done, let's go home.
+				if (!isturf(home))
+					explode()
+					return
+				if (get_turf(src) == get_turf(home)) return
+				src.navigate_to(get_turf(home), FLOORBOT_MOVE_SPEED, max_dist = 10)
+				if (!src.path || !length(src.path))
+					KillPathAndGiveUp(1)
+					return
+				return
+		// Something to do. Let's decide what to do first.
+		if (!target)
+			target = targets[1]
+		// Now we need to find out if we're going to move towards a prop dispenser or the table.
+		if (!moveTowards)
+			if (!hasDrink)
+				// if there's a barbuddy dispenser nearby, let's do the cute little animation thing. if not, use magic to summon a drink
+				for (var/obj/decal/fakeobjects/barbuddy_dispenser/D in view(5, src))
+					moveTowards = D
+				if (!moveTowards)
+					hasDrink = 1
+			else
+				moveTowards = targets[1]
+
+		if (IN_RANGE(get_turf(src), get_turf(src.moveTowards), 1))
+			bartend()
+			return
+
+		if (isnull(get_turf(moveTowards)))
```

Comment:
If `moveTowards` exists, `get_turf()` will always return a turf, so the following block will never trigger.

Reply:
It triggers when moveTowards is destroyed, which can happen by dismantling a table when the bot is about to put a drink on it. Just checking `isnull(moveTowards)` instead doesn't seem to work and I'm not sure how else I should be checking this.